src/main/java/org/tailormap/api/scheduling/IndexTask.java

```diff
@@ -0,0 +1,131 @@
+  private final FeatureSourceFactoryHelper featureSourceFactoryHelper;
+  private final SolrService solrService;
+  private final FeatureTypeRepository featureTypeRepository;
+  private final SearchIndexRepository searchIndexRepository;
+
+  private long indexId;
+  private String description;
+
+  public IndexTask(
+      @Autowired SearchIndexRepository searchIndexRepository,
+      @Autowired FeatureTypeRepository featureTypeRepository,
+      @Autowired FeatureSourceFactoryHelper featureSourceFactoryHelper,
+      @Autowired SolrService solrService) {
+
+    this.featureSourceFactoryHelper = featureSourceFactoryHelper;
+    this.solrService = solrService;
+    this.featureTypeRepository = featureTypeRepository;
+    this.searchIndexRepository = searchIndexRepository;
+  }
+
+  @Timed(value = "indexTask", description = "Time taken to execute index task")
+  @Override
+  protected void executeInternal(@NonNull JobExecutionContext context)
+      throws JobExecutionException {
+
+    final JobDataMap persistedJobData = context.getJobDetail().getJobDataMap();
+    // final long indexId = persistedJobData.getLong(INDEX_KEY);
+    logger.info(
+        "Start Executing IndexTask {} for index {}, described with '{}'",
+        context.getJobDetail().getKey(),
+        getIndexId(),
+        getDescription());
+
+    SearchIndex searchIndex =
+        searchIndexRepository
+            .findById(getIndexId())
+            .orElseThrow(() -> new JobExecutionException("Search index not found"));
+
+    TMFeatureType indexingFT =
+        featureTypeRepository
+            .findById(searchIndex.getFeatureTypeId())
+            .orElseThrow(() -> new JobExecutionException("Feature type not found"));
+
+    try (SolrClient solrClient = solrService.getSolrClientForIndexing();
+        SolrHelper solrHelper = new SolrHelper(solrClient)) {
+
+      searchIndex.setStatus(SearchIndex.Status.INDEXING);
+      searchIndex = searchIndexRepository.save(searchIndex);
+
+      searchIndex =
+          solrHelper.addFeatureTypeIndex(
+              searchIndex, indexingFT, featureSourceFactoryHelper, searchIndexRepository);
+      searchIndex = searchIndex.setStatus(SearchIndex.Status.INDEXED);
+      searchIndexRepository.save(searchIndex);
+      persistedJobData.put(
+          "executions", (1 + (int) context.getMergedJobDataMap().getOrDefault("executions", 0)));
+      persistedJobData.put("lastExecutionFinished", Instant.now());
+      persistedJobData.put("lastResult", "Index task executed successfully");
+      context.setResult("Index task executed successfully");
+    } catch (UnsupportedOperationException | IOException | SolrServerException | SolrException e) {
+      logger.error("Error indexing", e);
+      searchIndex.setStatus(SearchIndex.Status.ERROR).setComment(e.getMessage());
+      searchIndexRepository.save(searchIndex);
+      throw new JobExecutionException("Error indexing", e);
```

Comment:
When execution fails, is it possible to update the `context` or does this need to be done in a Listener?
So when making a UI for tasks, we can see the `lastResult` also for a failure.

Idea to add a POC task that fails?